App.tsx

```diff
@@ -1,19 +1,26 @@
 import 'localization';
-import React from 'react';
+import React, { useState } from 'react';
 import { StatusBar } from 'react-native';
 
 import { NavigationContainer, useTheme } from '@react-navigation/native';
 import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
 
+import AnimatedBootSplash from 'common/AnimatedSplash';
+
 import NavigationStack from 'navigation';
 
-import { useThemeConfig } from 'themes/hooks';
+import { useThemeConfig } from 'themes/useThemeConfig';
 
 const client = new QueryClient();
 
 const App = () => {
   const { theme } = useThemeConfig();
   const { dark: isDarkMode } = useTheme();
+  const [visible, setVisible] = useState(true);
+
+  if (visible) {
+    return <AnimatedBootSplash onAnimationEnd={() => setVisible(false)} />;
+  }
```

Comment:
Why not a `useEffect`?

Reply:
For which part exactly? I'm not sure how can this be moved to a `useEffect`.

I think that one of the reasons to not use a `useEffect` is that we want to render a component (`AnimatedBootSplash`)

Reply:
Ah, I think I got confused! Closing this comment